neuro_san/internals/network_providers/agent_network_storage.py

```diff
@@ -97,13 +97,15 @@ def remove_agent_network(self, agent_name: str):
         so that agent becomes unavailable on our server.
         """
         with self.lock:
+            agent_network: AgentNetwork = self.agents_table.get(agent_name, None)
             self.agents_table.pop(agent_name, None)
```

Comment:
Works. I think pop() will return you the value you want, i.e. this could be done in a single line.
But really, fine as-is.